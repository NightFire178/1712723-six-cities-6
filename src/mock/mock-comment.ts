import mockUser from "./mock-user";
import comment from '../types/comment'

const mockComment:comment = {
  comment: `Beautiful space, fantastic location and atmosphere,
  really a wonderful place to spend a few days. Will be back.`,
  date: `2021-03-07T08:04:28.647Z`,
  id: 10,
  rating: 4,
  user: mockUser
}
export default mockComment

export const mockArrayComments:Array<comment> =[
  {
    comment: `Beautiful space, fantastic location and atmosphere,
  really a wonderful place to spend a few days. Will be back.`,
    date: `2021-03-07T08:04:28.647Z`,
    id: 15,
    rating: 3,
    user: mockUser
  },
  {
    comment: `Beautiful space, fantastic location and atmosphere,
  really a wonderful place to spend a few days. Will be back.`,
    date: `2021-03-07T08:04:28.647Z`,
    id: 13,
    rating: 5,
    user: mockUser
  },
  {
    comment: `What an amazing view! The house is stunning and in
    an amazing location. The large glass wall had an amazing view of the river!`,
    date: `2021-03-07T08:04:28.647Z`,
    id: 10,
    rating: 4,
    user: mockUser
  }
]
