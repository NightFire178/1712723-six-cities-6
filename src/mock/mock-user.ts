import User from "../types/user";

const mockUser:User = {
  id: 1,
  name: `vasa`,
  is_pro: true,
  avatar_url:`https://assets.htmlacademy.ruâ€¦tar/6.jp`,
  email: `sdfhskd@maid.ru`
}
export default mockUser
