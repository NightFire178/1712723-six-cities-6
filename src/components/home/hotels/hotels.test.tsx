import React from 'react'
import {render} from '@testing-library/react'
import Hotels, {sortName} from "./hotels";
import {mockArrayHotels} from '../../../mock/mock-hotels'


const useDispatch = ()=>(jest.fn())

jest.mock(`react-redux`, ()=>({useDispatch}))

const appStateSelection = {
  sort: ()=>sortName.popular
}

jest.mock("../../../redux/selectors/app-state", ()=>appStateSelection)

//eslint-disable-next-line
jest.mock(`../../block/card/card`, ()=>()=><div>Card</div>)

test(`nav block component test`, () => {
  const hotels = JSON.parse(JSON.stringify(mockArrayHotels))
  const header = render(<Hotels hotels={hotels} /> )
  expect(header).toMatchSnapshot()
})
