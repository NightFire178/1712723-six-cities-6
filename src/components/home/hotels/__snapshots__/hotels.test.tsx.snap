// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hotels home component test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="cities__places places"
      >
        <h2
          class="visually-hidden"
        >
          Places
        </h2>
        <b
          class="places__found"
        >
          13
           places to stay in 
          Brussels
        </b>
        <form
          action="#"
          class="places__sorting"
          method="get"
        >
          <span
            class="places__sorting-caption"
          >
            Sort by
          </span>
          <span
            class="places__sorting-type"
            tabindex="0"
          >
            Popular
            <div>
              <svg
                class="places__sorting-arrow"
                height="4"
                width="7"
              >
                <use
                  xlink:href="#icon-arrow-select"
                />
              </svg>
            </div>
          </span>
          <ul
            class="places__options places__options--custom false"
          >
            <li
              class="places__option places__option--active"
              tabindex="0"
              value="0"
            >
              Popular
            </li>
            <li
              class="places__option"
              tabindex="0"
              value="1"
            >
              Price: low to high
            </li>
            <li
              class="places__option"
              tabindex="0"
              value="2"
            >
              Price: high to low
            </li>
            <li
              class="places__option"
              tabindex="0"
              value="3"
            >
              Top rated first
            </li>
          </ul>
        </form>
        <div
          class="cities__places-list places__list tabs__content"
        >
          <div>
            Card
          </div>
          <div>
            Card
          </div>
          <div>
            Card
          </div>
          <div>
            Card
          </div>
          <div>
            Card
          </div>
          <div>
            Card
          </div>
          <div>
            Card
          </div>
          <div>
            Card
          </div>
          <div>
            Card
          </div>
          <div>
            Card
          </div>
          <div>
            Card
          </div>
          <div>
            Card
          </div>
          <div>
            Card
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="cities__places places"
    >
      <h2
        class="visually-hidden"
      >
        Places
      </h2>
      <b
        class="places__found"
      >
        13
         places to stay in 
        Brussels
      </b>
      <form
        action="#"
        class="places__sorting"
        method="get"
      >
        <span
          class="places__sorting-caption"
        >
          Sort by
        </span>
        <span
          class="places__sorting-type"
          tabindex="0"
        >
          Popular
          <div>
            <svg
              class="places__sorting-arrow"
              height="4"
              width="7"
            >
              <use
                xlink:href="#icon-arrow-select"
              />
            </svg>
          </div>
        </span>
        <ul
          class="places__options places__options--custom false"
        >
          <li
            class="places__option places__option--active"
            tabindex="0"
            value="0"
          >
            Popular
          </li>
          <li
            class="places__option"
            tabindex="0"
            value="1"
          >
            Price: low to high
          </li>
          <li
            class="places__option"
            tabindex="0"
            value="2"
          >
            Price: high to low
          </li>
          <li
            class="places__option"
            tabindex="0"
            value="3"
          >
            Top rated first
          </li>
        </ul>
      </form>
      <div
        class="cities__places-list places__list tabs__content"
      >
        <div>
          Card
        </div>
        <div>
          Card
        </div>
        <div>
          Card
        </div>
        <div>
          Card
        </div>
        <div>
          Card
        </div>
        <div>
          Card
        </div>
        <div>
          Card
        </div>
        <div>
          Card
        </div>
        <div>
          Card
        </div>
        <div>
          Card
        </div>
        <div>
          Card
        </div>
        <div>
          Card
        </div>
        <div>
          Card
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
